include/swift/SIL/SILValue.h

```diff
@@ -128,24 +130,45 @@ struct ValueOwnershipKind {
   bool isTrivialOr(ValueOwnershipKind Kind) const {
     return Value == Trivial || Value == Kind;
   }
+
+  /// Given that there is an aggregate value (like a struct or enum) with this
+  /// ownership kind, and a subobject of type Proj is being projected from the
+  /// aggregate, return Trivial if Proj has trivial type and the aggregate's
+  /// ownership kind otherwise.
+  ValueOwnershipKind getProjectedOwnershipKind(SILModule &M,
+                                               SILType Proj) const;
 };
 
 llvm::raw_ostream &operator<<(llvm::raw_ostream &os, ValueOwnershipKind Kind);
 
 /// This is the base class of the SIL value hierarchy, which represents a
-/// runtime computed value. Things like SILInstruction derive from this.
+/// runtime computed value. Some examples of ValueBase are SILArgument and
+/// SingleValueInstruction.
 class ValueBase : public SILNode, public SILAllocated<ValueBase> {
+  friend class Operand;
+
   SILType Type;
   Operand *FirstUse = nullptr;
-  friend class Operand;
+
+  /// A 32 bit field of data that we provide for abstract subclasses to use.
+  uint32_t SubclassData;
```

Comment:
Why is this in the base class. Why not putting the info in MultipleValueInstructionResult?
@rjmccall maybe a question to you

Reply:
Because the space is already there in the base class because of alignment padding.

Reply:
In a follow on commit, I will add some static_asserts/comments that says this.

Reply:
Er, wait, I was assuming you would add this to SILNode.  It is not using spare bits when you add it to ValueBase like this.

Reply:
Ah, I read the first part of your previous comment where you were talking about putting this in the spare bits of ValueBase. For whatever reason I didn't grok the 2nd part of the sentence. I will fix this.